scio-test/src/main/scala/com/spotify/scio/testing/PipelineTestUtils.scala

```diff
@@ -155,6 +158,31 @@ trait PipelineTestUtils {
       fn(sc.parallelize(data1), sc.parallelize(data2), sc.parallelize(data3), sc.parallelize(data4))
     }._2
 
+  /**
+   * Test pipeline components with a [[ScioContext]] and replace transforms with the provided
+   * override.
+   *
+   * @param fn
+   *   code that tests the components and verifies the result
+   *
+   * {{{
+   * runWithOverrides(TransformOverride.of("operation", (v: Int) => v.toString)) { sc =>
+   *   val result = sc.parallelize(Seq(1, 2, 3))
+   *     .withName("operation")
+   *     .map(operationFn)
+   *   result should contain theSameElementAs("1", "2", "3")
+   * }
+   * }}}
+   */
+  def runWithOverrides(
```

Comment:
I tried but then it's not possible to decide when the pipeline should be run.

Reply:
Ahh yeah that makes sense. Could we nest it somehow?

```
runWithContext { sc =>
  sc.withOverrides(
    TransformOverride.of(...)
  )
}
```

Reply:
looks nice and should be possible. Will try that.

Reply:
This change is a bit risky because the transform must be defined before the `withOverrides` call or we need to store the override and call  `replaceAll` just before run.

I tried to integrate the `TestDataManager` here but it is inconvenient